_search-plugins/search-pipelines/neural-sparse-query-two-phase-processor.md

```diff
@@ -0,0 +1,124 @@
+`two_phase_parameter.max_window_size` | Int | A limit number of the two-phase fine-tune documents. Default value is 10000. Valid range is greater than 50.
+`tag` | String | The processor's identifier. Optional.
+`description` | String | A description of the processor. Optional.
+
+## Example
+
+### Create search pipeline
+
+The following example request creates a search pipeline with a `neural_sparse_two_phase_processor` search request processor. The processor sets a custom model ID at the index level and provides different default model IDs for two specific fields in the index:
+
+```json
+PUT /_search/pipeline/two_phase_search_pipeline
+{
+  "request_processors": [
+    {
+      "neural_sparse_two_phase_processor": {
+        "tag": "neural-sparse",
+        "description": "This processor is making two-phase processor.",
+        "enabled": true,
+        "two_phase_parameter": {
+          "prune_ratio": custom_prune_ratio,
+          "expansion_rate": custom_expansion_rate,
+          "max_window_size": custom_max_window_size
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Set search pipeline
+
+Then choose the proper index and set the `index.search.default_pipeline` to the pipeline name.
+```json
+PUT /index-name/_settings 
+{
+  "index.search.default_pipeline" : "two_phase_search_pipeline"
+}
+```
+{% include copy-curl.html %}
+
+## Limitation
+### Version support
+`neural_sparse_two_phase_processor` is introduced in OpenSearch 2.15. You can use this pipeline in a cluster whose minimal version is greater than or equals to 2.15.
+
+### Compound query support
+There is 6 types of [compound query]({{site.url}}{{site.baseurl}}/query-dsl/compound/index/). And we only support bool query now.
+- [x] bool (Boolean)
+- [ ] boosting 
+- [ ] constant_score
+- [ ] dis_max (disjunction max)
+- [ ] function_score
+- [ ] hybrid
+
+Notice, neural sparse query or bool query with a boost parameter (not same as boosting query) are also supported.
+
+#### Supported Example
+##### Single neural sparse query
+
+```
+GET /my-nlp-index/_search
+{
+  "query": {
+    "neural_sparse": {
+      "passage_embedding": {
+        "query_text": "Hi world"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+##### Neural sparse query nested in bool query
+
+```
+GET /my-nlp-index/_search
+{
+  "query": {
+    "bool": {
+      "should": [
+        {
+          "neural_sparse": {
+            "passage_embedding": {
+              "query_text": "Hi world",
+              "model_id": <model-id>
+            },
+            "boost": 2.0
+          }
+        }
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+  
+## Metrics
+
+In doc-only mode, the two-phase processor will reduce the query latency by 20% to 50%, depending on the index configuration and two-phase parameters.
```

Comment:
`also depending on the index configuration and two-phase parameters` Do you mean this depends on the dataset distribution? Because I think we're using default two-phase parameters and consistent index settings in our experiments

Reply:
> is this sparse_vector search or end to end latency

end to end.

Reply:
> `also depending on the index configuration and two-phase parameters` Do you mean this depends on the dataset distribution? Because I think we're using default two-phase parameters and consistent index settings in our experiments

Yes, i corrected the description to index specific data distribution and dataset size .

Reply:
> end to end.

I think the speed up ratio on pure sparse vector is more important. The model inference cost and retrieval cost is orthogonal and the 2-phase pipeline only optimize the retrieval cost.